File: src/main/php/Easy/FindLargestAltitude.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class FindLargestAltitude
{
    /**
     * @param  int[]  $gain
     */
    public function solve(array $gain): int
    {
        $result = array_fill(0, count($gain) + 1, 0);
        $altitude = 0;
        for ($i = 0; $i < count($gain); $i++) {
            $altitude += $gain[$i];
            $result[$i] = $altitude;
        }

        return max($result);
    }
}

File: src/easy/FindLargestAltitude.swift
func findLargestAltitude(_ gain: [Int]) -> Int {
  var result = Array(repeating: 0, count: gain.count + 1)
  var altitude = 0
  for i in gain.indices {
    result[i] = gain[i] + altitude
    altitude += gain[i]
  }
  return result.max() ?? 0
}
